### PR TITLE
refactor run_sorter() with docker integration

### PR DESCRIPTION
**Pull request by samuelgarcia**

### CODE REVIEW

**Comment by samuelgarcia** [comment]
@alejoe91 @bendichter : this is a dead simple and working implementation of running a sorters into a docker using the python `docker` package.

The only constrain is that the docker need to have spiekinterface 0.90.0 inside which is possible only with source at the moment.
After July, we will be able to made collection od docker image with it.

Compared to the version using `hither` done by Alessio (in the old API), here the `docker_image='...'` can be change on demand and is not hard coded in sources.
This will be super convinient to test differents version of the same sorter within the same code.

Please make some feedback.

@alejoe91  : If you validate, I will made a small doc also.

**Comment by samuelgarcia** [comment]
@alejoe91 : if you have 3 minuts tomorow, could you approve this PR. So we can merge.
I will need to make demo soon.
We will be able to improve docker stuff later.
Thanks a lot.

File: spikeinterface/sorters/runsorter.py [hunk 1]
```diff
@@ -52,85 +77,226 @@ def run_sorter(sorter_name, recording, output_folder=None,
...
     
     if delete_output_folder:
-        raise NotImplementedError
-        # TODO : delete_output_folder
+        shutil.rmtree(output_folder)
     
     return sorting
 
 
+def modify_input_folder(rec_dict, input_folder):
```
Comment (alejoe91) [inline spikeinterface/sorters/runsorter.py:85]: @samuelgarcia this should also handle lists of paths (see the `relative_to` in base.py)

File: spikeinterface/sorters/runsorter.py [hunk 2]
```diff
@@ -52,85 +77,226 @@ def run_sorter(sorter_name, recording, output_folder=None,
...
+            raise Exception
+
+
+def run_sorter_docker(sorter_name, recording, docker_image, output_folder=None,
+            remove_existing_folder=True, delete_output_folder=False,
+            verbose=False, raise_error=True, **sorter_params):
+    
+    import docker
+    
+    assert output_folder is not None, 'with run in docker output_folder must be specify'
```
Comment (alejoe91) [inline spikeinterface/sorters/runsorter.py:119]: We could also set the same default folder as in the local version

**Review by alejoe91 — CHANGES_REQUESTED** [review]
@samuelgarcia this looks great! Just have a couple of comments

**Comment by samuelgarcia** [comment]
Thank you for review.
Changes done.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**